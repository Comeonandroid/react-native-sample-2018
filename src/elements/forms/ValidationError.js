import React from 'react'
import styled from 'styled-components'
import { prop } from 'styled-tools'
import { PropTypes } from 'prop-types'

import { Text } from '../index'


const ValidationError = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

ValidationError.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  color: ${ prop('theme.colors.danger')};
  font-size: 12px;
`

export default ValidationError