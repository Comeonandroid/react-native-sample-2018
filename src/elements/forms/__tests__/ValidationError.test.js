import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import ValidationError from '../ValidationError'

describe('<ValidationError />', () => {
  itBehavesLike('aThemedComponent', { subject: <ValidationError /> })
  itBehavesLike('aComponentWithChildren', { subject: <ValidationError /> })
})
