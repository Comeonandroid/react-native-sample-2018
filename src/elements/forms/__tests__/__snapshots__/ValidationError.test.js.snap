// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationError /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fcfcfc",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#f25c74",
          "fontSize": 12,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ValidationError /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#f25c74",
          "fontSize": 12,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<ValidationError /> renders component when passed in 1`] = `
<Styled(Text)>
  <Component />
</Styled(Text)>
`;

exports[`<ValidationError /> renders text when passed in 1`] = `
<Styled(Text)>
  Test
</Styled(Text)>
`;
